packages/workspace/src/browser/workspace-storage-service.ts

```diff
@@ -18,36 +20,40 @@ import { LocalStorageService } from '@theia/core/lib/browser/storage-service';
 export class WorkspaceStorageService implements StorageService {
 
     private prefix: string;
-    private initialized: Promise<void>;
+    protected initialized: Deferred<void>;
     protected storageService: StorageService;
 
-    constructor( @inject(WorkspaceService) protected workspaceService: WorkspaceService,
-        @inject(ILogger) protected logger: ILogger) {
-        this.initialized = this.workspaceService.root.then(stat => {
-            if (stat) {
-                this.prefix = stat.uri;
-            } else {
-                this.prefix = '_global_';
-            }
-        });
+    @inject(WorkspaceService)
+    protected workspaceService: WorkspaceService;
+
+    @inject(ILogger)
+    protected logger: ILogger;
+
+    constructor() {
         this.storageService = new LocalStorageService(this.logger);
+        this.initialized = new Deferred<void>();
+    }
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        const statFile = await this.workspaceService.root;
+        this.prefix = statFile ? new URI(statFile.uri).path.toString() : '_global_';
```

Comment:
`new URI(statFile.uri).path.toString()` is different from `stat.uri`, is it on purpose?

Could you rollback changes which are out of scope of this PR that we don't break anything?

Reply:
```ts
> uri.toString()
'file:///etc/test/'
> uri.path.toString()
'/etc/test/'
>
```
It was to remove the scheme, but yes I can.